File: backend/src/main/java/kr/co/kalpa/pcms/domain/movie/entity/Movie.java
package kr.co.kalpa.pcms.domain.movie.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Movie {
    private Long id;
    private String mid;
    private String gubun;
    private String title1;
    private String title2;
    private String title3;
    private String category;
    private String gamdok;
    private String makeYear;
    private String nara;
    private String dvdId;
    private String title1num;
    private String title1title2;
}

File: frontend/src/domain/movie/types/movie.ts
export interface MovieDto {
  id?: number;
  mid?: string;
  gubun: string;
  title1: string;
  title2?: string;
  title3?: string;
  category?: string;
  gamdok?: string;
  makeYear?: string;
  nara?: string;
  dvdId?: string;
  title1num?: string;
  title1title2?: string;
}

export interface MovieSearchDto {
  page?: number;
  size?: number;
  keyword?: string;
  category?: string;
  gamdok?: string;
  makeYear?: string;
  gubun?: string;
  nara?: string;
}

export interface MovieReviewDto {
  id?: number;
  title: string;
  nara?: string;
  year?: string;
  lvl?: number;
  ymd?: string;
  content?: string;
  lastmodifyDt?: string;
}

export interface MovieReviewSearchDto {
  page?: number;
  size?: number;
  keyword?: string;
  startYmd?: string;
  endYmd?: string;
  minLvl?: number;
}

export interface HddDto {
  id?: number;
  volumnName?: string;
  gubun: string;
  path?: string;
  fileName?: string;
  name: string;
  pdir?: string;
  extension?: string;
  size?: number;
  sha1Cd?: string;
  srchKey?: string;
  lastModifiedYmd?: string;
  pid?: number;
  rightPid?: number;
}

export interface HddSearchDto {
  page?: number;
  size?: number;
  keyword?: string;
  volumnName?: string;
  extension?: string;
  gubun?: string;
}
